lib/SILOptimizer/Mandatory/SILGenCleanup.cpp

```diff
@@ -0,0 +1,78 @@
+//===--- SILGenCleanup.cpp ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// Perform peephole-style "cleanup" to aid SIL diagnostic passes.
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "silgen-cleanup"
+
+#include "swift/SIL/SILInstruction.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+#include "swift/SILOptimizer/Utils/CanonicalizeInstruction.h"
+
+using namespace swift;
+
+// Define a CanonicalizeInstruction subclass for use in SILGenCleanup.
+struct SILGenCanonicalize : CanonicalizeInstruction {
+  bool changed = false;
+
+  SILGenCanonicalize() : CanonicalizeInstruction(DEBUG_TYPE) {}
+
+  void genInstruction(SILInstruction *) { changed = true; }
+
+  void killInstruction(SILInstruction *inst) {
+    inst->eraseFromParent();
+    changed = true;
+  }
+
+  void newUsers(SILValue) { changed = true; }
+};
+
+//===----------------------------------------------------------------------===//
+//                            Top Level Pass
```

Comment:
We generally have been using Top Level Entrypoint. Also, are you sure that Top Level Pass is centered? Here is an emacs macro that does this automagically if you need:

```
(defun center-banner ()
  "Create a centered banner using the data on the current line."
  (interactive)
  (let* ((slash-string (make-string 2 ?/))
         (equal-string (make-string 3 ?=))
         (banner-string (concat slash-string
                                equal-string
                                (make-string 70 ?-)
                                equal-string
                                slash-string)))
    (save-excursion
      ;; Create banner on previous line.
      (forward-line -1)
      (end-of-line)
      (newline)
      (insert banner-string)
      (forward-line 1)
      (let ((lm (current-left-margin))
            line-length)
        (beginning-of-line)
        (delete-horizontal-space)
        (end-of-line)
        (delete-horizontal-space)
        (setq line-length (current-column))
        (if (> (- fill-column lm line-length 2) 0)
            (indent-line-to
             (+ lm (- (/ (- fill-column lm line-length) 2) 2))))
        (beginning-of-line)
        (insert-before-markers slash-string)
        (end-of-line)
        (newline)
        (insert banner-string)))))
```